src/ts/form_render/formRender.ts

```diff
@@ -0,0 +1,365 @@
+   */
+  // renderControl(element = null) {
+  //   const opts = this.options
+  //   const fieldData = opts.formData
+  //   if (!fieldData || Array.isArray(fieldData)) {
+  //     throw new Error(
+  //       'To render a single element, please specify a single object of formData for the field in question',
+  //     )
+  //   }
+  //   const sanitizedField = this.santizeField(fieldData)
+
+  //   // determine the control class for this type, and then build it
+  //   const engine = new opts.layout()
+  //   const controlClass = control.getClass(fieldData.type, fieldData.subtype)
+  //   const forceTemplate = opts.forceTemplate || 'hidden' // support the ability to override what layout template the control is rendered using. This can be used to output the whole row (including label, help etc) using the standard templates if desired.
+  //   const field = engine.build(controlClass, sanitizedField, forceTemplate)
+  //   element.appendFormFields(field)
+  //   opts.notify.success(opts.messages.formRendered)
+  //   return this
+  // }
+
+  /**
+   * Return user entered data
+   */
+  get userData() {
+    const options = this.options
+    const definedFields = options.formData.slice()
+
+    // save tinyMCE editors
+    definedFields
+      .filter(fieldData => fieldData.subtype === 'tinymce')
+      .forEach(fieldData => window.tinymce.get(fieldData.name).save({}))
+
+    this.instanceContainers.forEach(container => {
+      const userDataMap = $('select, input, textarea', container)
+        .serializeArray()
+        .reduce((acc, { name, value }) => {
+          name = name.replace('[]', '')
+          if (acc[name]) {
+            acc[name].push(value)
+          } else {
+            acc[name] = [value]
+          }
+          return acc
+        }, {})
+
+      const definedFieldsLength = definedFields.length
+      for (let i = 0; i < definedFieldsLength; i++) {
+        const definedField = definedFields[i]
+        // Skip fields that have no name--Likely these are fields that do not hold data(h1,p)
+        if (definedField.name === undefined) continue
+        // Skip disabled fields -- This will not have user data available
+        if (definedField.disabled) continue
+
+        definedField.userData = userDataMap[definedField.name]
+      }
+    })
+
+    return definedFields
+  }
+
+  clearFormData() {
+    this.instanceContainers.forEach(container => {
+      // clear tinyMCE editors
+      this.options.formData
+        .slice()
+        .filter(fieldData => fieldData.subtype === 'tinymce')
+        .forEach(fieldData => window.tinymce.get(fieldData.name).setContent(''))
+
+      container.querySelectorAll('input, select, textarea').forEach(input => {
+        if (['checkbox', 'radio'].includes(input.type)) {
+          input.checked = false
+        } else {
+          input.value = ''
+        }
+      })
+    })
+  }
+
+  //Kevin -- how is this method different than helpers.ts getData() ?
```

Comment:
they look pretty similar. can probably be merge

Reply:
@kevinchappell can you attempt this